ntbs-service-unit-tests/DataMigration/ImportValidatorContactTracingTest.cs

```diff
@@ -0,0 +1,505 @@
+                AdultsActiveTB = 0,
+                AdultsLatentTB = 5,
+                AdultsStartedTreatment = 5,
+                AdultsFinishedTreatment = 5,
+
+                ChildrenIdentified = 5,
+                ChildrenScreened = 4,
+                ChildrenActiveTB = 1,
+                ChildrenLatentTB = 3,
+                ChildrenStartedTreatment = 2,
+                ChildrenFinishedTreatment = 1,
+            };
+
+            // Act
+            var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
+
+            // Assert
+            // Validation errors
+            var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            // Data cleaning warnings
+            _importLoggerMock
+                .Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId, It.IsAny<string>()),
+                    Times.Never);
+
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsIdentified), 5);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsScreened), 5);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsActiveTB), 0);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsLatentTB), 5);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsStartedTreatment), 5);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsFinishedTreatment), 5);
+
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenIdentified), 5);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenScreened), 4);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenActiveTB), 1);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenLatentTB), 3);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenStartedTreatment), 2);
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenFinishedTreatment), 1);
+        }
+
+        private void AssertWarningsGiven(Notification notification, IEnumerable<string> warnings)
+        {
+            foreach (var warning in warnings)
+            {
+                _importLoggerMock.Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId, warning));
+            }
+            _importLoggerMock.VerifyNoOtherCalls();
+        }
+
+        private static void AssertContactTracingValuePreserved(Notification notification, string propertyName,
+            int? value)
+        {
+            Assert.Equal(value,
+                notification.ContactTracing.GetType().GetProperty(propertyName)?.GetValue(notification.ContactTracing));
```

Comment:
Since you already know the type, it's better to use `typeof(ContactTracing)` rather than `notification.ContactTracing.GetType()`.

I think this happens in your production code too.

Reply:
Thanks for spotting that! Fixed